validator_worker/src/sentry_interface.rs

```diff
@@ -26,23 +28,70 @@ pub struct SentryApi<T: Adapter> {
     pub propagate_to: Vec<(ValidatorDesc, String)>,
 }
 
-impl<T: Adapter + 'static> SentryApi<T> {
+#[derive(Debug)]
+pub enum Error<AE: AdapterErrorKind> {
+    BuildingClient(reqwest::Error),
+    Request(reqwest::Error),
+    ValidatorAuthentication(AdapterError<AE>),
+    MissingWhoamiInChannelValidators {
+        channel: ChannelId,
+        validators: Vec<ValidatorId>,
+        whoami: ValidatorId,
+    },
+}
+
+impl<AE: AdapterErrorKind> std::error::Error for Error<AE> {}
+
+impl<AE: AdapterErrorKind> fmt::Display for Error<AE> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use Error::*;
+
+        match self {
+            BuildingClient(err) => write!(f, "Building client: {}", err),
+            Request(err) => write!(f, "Making a request: {}", err),
+            ValidatorAuthentication(err) => {
+                write!(f, "Getting authentication for validator: {}", err)
```

Comment:
We should make these simpler,  this is quite okay
```js
BuildingClient(err) => write!(f, "BuildingClient: {}", err),
            Request(err) => write!(f, "Request: {}", err),
            ValidatorAuthentication(err) => {
                write!(f, "ValidatorAuthentication: {}", err)
```

Reply:
This is only for Display (i.e. printing out) we use the Debug impl for the logging so this should be fine for reading